### PR TITLE
Fix the statistics initializations of eigendecompose Shampoo and eigenvalue-corrected Shampoo

### PR DESCRIPTION
**Pull request by tsunghsienlee**
Summary:
This diff fixes the statistics initializations of eigendecompose Shampoo and eigenvalue-corrected Shampoo. For eigendecompose Shampoo, this changes include initializing the eigenvectors matrices as identity matrices and initializing the eigenvalues as all ones. For eigenvalue-corrected Shampoo, this changes include initializing the eigenvectors matcies as identity matrices, and keeping corrected eigenvalues as all 0s.

Extend `distributed_state_dict()`-related tests to include testing eigendecompose Shampoo and eigenvalue-corrected Shampoo to verify the initializations.

Differential Revision: D72093421

### CODE REVIEW

File: distributed_shampoo/utils/shampoo_preconditioner_list.py [hunk 1]
```diff
@@ -1147,6 +1151,9 @@ def _create_kronecker_factors_state_for_block(
             )
             for dim in preconditioned_dims
         )
+        # Initialize factor_matrices_eigenvalues all ones.
+        for t in factor_matrices_eigenvalues:
+            block_info.get_tensor(t).fill_(1.0)
```
Comment (runame) [inline distributed_shampoo/utils/shampoo_preconditioner_list.py:1156]: This could also be implemented with a new `block_info.allocate_ones_tensor`method.

File: distributed_shampoo/utils/shampoo_preconditioner_list.py [hunk 2]
```diff
@@ -1139,6 +1139,10 @@ def _create_kronecker_factors_state_for_block(
             )
             for dim in preconditioned_dims
         )
+        # Initialize factor_matrices_eigenvectors as identity matrices.
+        for t in factor_matrices_eigenvectors:
+            block_info.get_tensor(t).fill_diagonal_(1.0)
+
```
Comment (runame) [inline distributed_shampoo/utils/shampoo_preconditioner_list.py:1145]: This could be refactored as a new `block_info.allocate_eye_tensor` method that calls `block_info.allocate_zeros_tensor` followed by this code.

The advantage is a unified interface for all initialization options (`zeros`, `ones`, `eye`) and it will be easier to change the implementation of each initialization method later on.
